### PR TITLE
Feature/deprecate embed block

### PR DESCRIPTION
**Pull request by divyarajmasani**
## Context

With the addition of the new Web Stories block, the existing Stories Embed block needs to be deprecated and handle the block transformation.

## Summary

This PR removes the Stories Embed block and adds appropriate transform and deprecation definition for the new block to handle older content.

- Technically this PR updates the Stories Embed block with new block types, latest and selected stories.
- These newly added block types enables users to embed multiple stories on a page.

## Relevant Technical Choices

- Removes existing files relating to the Stories Embed block.
- Renames Web Stories List to Stories Embed block.

## To-do

- Update any relevant documentation for the Stories block.
- Expand test coverage for the new block.

## User-facing changes

- Users may observe changes in older embed block which now will have transformed into newer block.

## Testing Instructions

- Test new web stories block like any other block, add it to any post and try different settings.

Test with older content.
- Create a new post with older block content, either using the classic editor's 'text' view or block editors 'Code editor' and save the post. Following is the snippet of older block.

```
<!-- wp:web-stories/embed {"url":"https://preview.amp.dev/documentation/examples/introduction/stories_in_amp","title":"Stories in AMP - Hello World","poster":"https://amp.dev/static/samples/img/story_dog2_portrait.jpg"} -->
<div class="wp-block-web-stories-embed alignnone"><a href="https://preview.amp.dev/documentation/examples/introduction/stories_in_amp"><img src="https://amp.dev/static/samples/img/story_dog2_portrait.jpg" alt="Stories in AMP - Hello World" width="360" height="600" /></a></div>
<!-- /wp:web-stories/embed -->
```

- Now, switch to the block editor and refresh once.
- Check that the block toolbar now has a transform button to switch between blocky types.

Test 'Convert to blocks' with classic block and shortcode.
- Add following shortcode to a classic block.
```
[web_stories_embed align="none" height="457" width="285" poster="https://amp.dev/static/samples/img/story_dog2_portrait.jpg" title="Test Story" url="https://preview.amp.dev/documentation/examples/introduction/stories_in_amp" /]
```
- Click the button which says 'Convert to blocks' on the classic block toolbar.
- This should transform the shortcode to web stories block without any errors.

Test content added in classic editor.
- Use [Classic Editor](https://wordpress.org/plugins/classic-editor/) plugin for this test.
- Create new post with older content, use above provided content.
- Now open the same post with block editor and use the 'Convert to blocks' on the classic content.
- This should transform the content to new block, without any errors.

### UAT

- [x] UAT should use the same steps as above.

### CODE REVIEW

File: assets/src/web-stories-block/block/edit.js [hunk 1]
```diff
@@ -45,7 +45,13 @@ import { icon } from './';
...
 
 function WebStoriesEdit({ attributes, setAttributes, className, isSelected }) {
-  const { blockType, viewType } = attributes;
+  const { blockType, viewType, url } = attributes;
+
+  if (!blockType && url && url.length) {
+    setAttributes({
+      blockType: BLOCK_TYPE_URL,
+    });
+  }
```
Comment (swissspidy) [inline assets/src/web-stories-block/block/edit.js:54]: Would something like this work?

```js
// If block type is not set, but there is a URL,
// this block was likely migrated from its previous form.
if (!blockType && url?.length) {
  return (
    <ConfigProvider config={config}>
      <StoryEmbedEdit
        icon={icon}
        attributes={attributes}
        setAttributes={setAttributes}
        className={className}
        isSelected={isSelected}
      />
    </ConfigProvider>
  );
}
```

Maybe together with an `useEffect` to set the blockType?

```js
useEffect(() => {
  if (!blockType && url?.length) {
    setAttributes({
      blockType: BLOCK_TYPE_URL,
    });
  }
}, [blockType, url, setAttributes]);
```

Comment (swissspidy) [reply assets/src/web-stories-block/block/edit.js]: I see you added the `useEffect` part.

Does this still cause this "flash" when first loading the block? Does the above `if (!blockType && url?.length) { return ( .... ) }` prevent this?

Comment (divyarajmasani) [reply assets/src/web-stories-block/block/edit.js]: The flash is still there. I'm looking into the docks and Gutenberg source if I find anything will update it. 
Also, I did remove Embed block files and moved the required ones to the new block directory.

Comment (swissspidy) [reply assets/src/web-stories-block/block/edit.js]: I thought with my `if (!blockType && url?.length) { return ( ... ) }` suggestion above this flash wouldn't be happening. Have you tried it?

Comment (divyarajmasani) [reply assets/src/web-stories-block/block/edit.js]: Update the deprecation definition and now it works, apparently, I needed to add `isEligible` for the migrate to work.

https://github.com/rtCamp/web-stories-wp/blob/5b21364cceb58fdf12470502397d1519298ca31d/assets/src/web-stories-block/block/deprecated.js#L110

and removed that `useEffect()` which was manually setting the `blockType`.

Comment (swissspidy) [reply assets/src/web-stories-block/block/edit.js]: Nice!!

**Review by swissspidy — APPROVED** [review]

**Review by spacedmonkey — APPROVED** [review]
SGTM

**Pull request merged**